### PR TITLE
ci(require-label): add new reusable workflow

### PR DESCRIPTION
**Pull request by xmfcx**
## Description

🤔 We have [make-sure-label-is-present.yaml](https://github.com/autowarefoundation/autoware-github-actions/blob/main/.github/workflows/make-sure-label-is-present.yaml) but it skips when it cannot find the label and let's the remaining actions decide what to do with the result. Because it doesn't fail, it skips the remaining workflows.

🫤 We also have [prevent-no-label-execution.yaml](https://github.com/autowarefoundation/autoware-github-actions/blob/main/.github/workflows/prevent-no-label-execution.yaml) which has the identical behavior. But this one is only here for legacy reasons, it can get confused when multiple labels are added at the same time.

So we needed a workflow that actually fails when the input label is not present.

`require-label` workflow does just that. It's a slightly modified version of [make-sure-label-is-present.yaml](https://github.com/autowarefoundation/autoware-github-actions/blob/main/.github/workflows/make-sure-label-is-present.yaml) but with additional summary and useful debug output.

### Failure

- https://github.com/autowarefoundation/autoware.universe/actions/runs/12470261679

image

image

### Success

- https://github.com/autowarefoundation/autoware.universe/actions/runs/12469760090

image

image

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: .github/workflows/require-label.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      label:
+        required: true
+        type: string
+    outputs:
+      result:
+        value: ${{ jobs.require-label.outputs.result }}
+
+jobs:
+  require-label:
+    runs-on: ubuntu-22.04
```
Comment (mitsudome-r) [inline .github/workflows/require-label.yaml:15]: Maybe we could use ubuntu-latest for this if there are no particular reasons for using ubuntu-22.04

**Review by sasakisasaki — APPROVED** [review]
### Quick Comment
Looks good!

### Reason for Approval
The workflow has two main features: success and failure cases. The PR description shows that the both features are tested well.

File: .github/workflows/require-label.yaml (hunk 1 above)
Comment (xmfcx) [reply .github/workflows/require-label.yaml]: Done.

**Comment by xmfcx** [comment]
Applied ubuntu-latest and tested on the last job:
 
image

**Pull request merged**